share/spice/gulp/gulp.js

```diff
@@ -0,0 +1,70 @@
+(function (env) {
+    "use strict";
+    
+    env.ddg_spice_gulp = function(api_result) {
+        
+        if (!api_result || !api_result.total) {
+            return Spice.failed('gulp');
+        }
+
+        // Get the original query sans the trigger words.
+        var script = $('[src*="/js/spice/gulp/"]')[0],
+            source = $(script).attr("src"),
+            query = source.match(/gulp\/([^\/]*)/)[1];
+        
+        var query_re = new RegExp(query, 'i');
+        
+        DDG.require('moment.js', function() {
+            Spice.add({
+                id: "grunt",
+                name: "Gulp Plugins",
+                data: api_result.results,
+                meta: {
+                    sourceName: 'Gulp',
+                    // We can't link to the actual query since filtering is all done on 
+                    // the front-end as well.
+                    sourceUrl: 'http://gulpjs.com/plugins/'
+                },
+                normalize: function(item) {
+                    // Make sure we have everything we need.
+                    if(!item.name.length || !item.version.length || !item.keywords.length ||
+                       !item.description.length || !item.rating.length || !item.modified.length) {
+                        return;
+                    }
+                    item.license = item.license || [];
+                    
+                    
+                    // Make sure that the query even exists in the name or description.
+                    if(!query_re.test(item.name[0]) && !query_re.test(item.description[0]) && 
+                       !query_re.test(item.keywords.toString())) {
+                        return;
+                    }
```

Comment:
@jagtalon No problem! To be honest I'm not the biggest fan of the current implementation of relevancy blocks in Spices!